contracts/CodexTitleMetadata.sol

```diff
@@ -103,11 +95,16 @@ contract CodexTitleMetadata is ERC721Token {
    * @return CodexTitleData token data for the given token ID
    */
   function getTokenById(uint256 _tokenId) public view
-    returns (bytes32 nameHash, bytes32 descriptionHash, bytes32[] imageHashes)
+    returns (bytes32 nameHash, bytes32 descriptionHash, bytes32[] fileHashes)
   {
-    CodexTitleData storage codexTitle = tokenData[_tokenId];
```

Comment:
Instead of storing a reference to `tokenData[_tokenId]`, I've changed it to read directly from `tokenData[_tokenId]`. I believe (possibly naively) that this would incur slightly smaller gas costs.

I also added a `require(exists(_tokenId))`, which seems to be a good idea.

Reply:
Might save a bit of gas if this is called on-chain (one would hope that the compiler would optimize things like this by default by I don't think it's quite there yet), although I think adding the require statement probably offsets those :).